src/drive/web/modules/actions/utils.js

```diff
@@ -0,0 +1,159 @@
+import { models } from 'cozy-client'
+import { isMobileApp, isIOS } from 'cozy-device-helper'
+import Alerter from 'cozy-ui/transpiled/react/Alerter'
+import { isReferencedByAlbum } from 'drive/web/modules/drive/files' // TODO move to cozy-client models
+import { forceFileDownload } from 'cozy-stack-client/dist/utils'
+import {
+  saveFileWithCordova,
+  saveAndOpenWithCordova
+} from 'drive/mobile/lib/filesystem'
+
+const { file: fileModel } = models
+const { isDirectory } = fileModel
+
+export const isAnyFileReferencedByAlbum = files => {
+  for (let i = 0, l = files.length; i < l; ++i) {
+    if (isReferencedByAlbum(files[i])) return true
+  }
+  return false
+}
+
+const isMissingFileError = error => error.status === 404
+
+const downloadFileError = error => {
+  return isMissingFileError(error)
+    ? 'error.download_file.missing'
+    : 'error.download_file.offline'
+}
+
+const openFileDownloadError = error => {
+  return isMissingFileError(error)
+    ? 'mobile.error.open_with.missing'
+    : 'mobile.error.open_with.offline'
+}
+
+export const downloadFiles = async (files, client) => {
```

Comment:
I would prefer that as a convention, for functions that need to be passed a client, we put `client` as the first argument. I already do this for all the code I write, and I think it helps since argument order here is not obvious.